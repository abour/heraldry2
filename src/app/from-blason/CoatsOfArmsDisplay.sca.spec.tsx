import { snapshotTest } from './tests/CoatsOfArms.render';

let value = 1;
jest.mock('../../utils/uuid', () => {
  return {
    uuid: () => value++,
  };
});

// It should render arms from http://blog.wirelizard.ca/2017/10/31/sca-field-only-armoury-project-part-one/

describe('CoatsOfArms', () => {
  beforeEach(() => {
    value = 0;
  });

  snapshotTest('Argent, chapé ployé per pale gules and sable');
  snapshotTest('Azure, chaussé ployé per pale gules and sable fimbriated or');
  snapshotTest('Barry and per chevron throughout pean and erminois');
  snapshotTest('Barry and per pale sable and argent, chaussé gules');
  snapshotTest('Barry azure and ermine');
  snapshotTest('Barry dancetty argent and azure');
  snapshotTest('Barry ermine and vert');
  snapshotTest('Barry of six or and sable');
  snapshotTest('Barry wavy ermine and sable');

  snapshotTest('Barry wavy vert and argent ermined vert');
  snapshotTest('Bendy and per bend sinister sable and Or');
  snapshotTest('Bendy and per pale sable and argent');
  snapshotTest('Bendy ermine and sable');
  snapshotTest('Bendy or and gules');
  snapshotTest('Bendy sinister and per bend azure and ermine');
  snapshotTest('Bendy sinister embattled or and gules');
  //snapshotTest('Bendy sinister of four, vert, argent, purpure and argent');
  snapshotTest('Bendy sinister sable and gules');

  snapshotTest('Bendy sinister vert and erminois');
  snapshotTest('Bendy wavy argent and sable');
  snapshotTest('Checky sable and ermine');
  //snapshotTest('Chevronelly azure and argent');
  //snapshotTest('Chevronelly erminois and pean');
  //snapshotTest('Chevronelly inverted gules and or');
  //snapshotTest('Chevronelly inverted vert and or');
  //snapshotTest('Gules scaly or');
  //snapshotTest('Gules mailed or');

  //snapshotTest('Gyronny arrondi argent voided sable and sable');
  //snapshotTest('Gyronny arrondy of six gules and argent');
  //snapshotTest('Gyronny arrondy Or and azure');
  //snapshotTest('Gyronny azure and argent scaly sable');
  snapshotTest('Gyronny erminois and pean');
  //snapshotTest('Gyronny gules and lozengy azure and or');
  //snapshotTest('Gyronny of ten ermine and vert');
});
