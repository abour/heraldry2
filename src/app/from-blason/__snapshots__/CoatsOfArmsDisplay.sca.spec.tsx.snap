// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoatsOfArms should render 'Argent, chapé ployé per pale gules and sable' 1`] = `
<svg
  className="coats-of-arms-display"
  height={266.6666666666667}
  viewBox="0 0 200 266.6666666666667"
  width={200}
>
  <defs>
    
    
    
    
    
    
    
    
    
    
    
    
    <clipPath
      id="plain-field-clip-path"
    >
      <path
        d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
        fill="transparent"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        stroke="#333"
        strokeWidth={1}
        style={
          Object {
            "cursor": "pointer",
            "pointerEvents": "none",
          }
        }
      />
    </clipPath>
  </defs>
  <g
    className="blason-field"
    clipPath="url(#plain-field-clip-path)"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <rect
      fill="#FFFFFF"
      height={266.6666666666667}
      stroke="#333"
      width={200}
      x={0}
      y={0}
    />
  </g>
  <g
    className="blason-ordinary"
    clipPath="url(#plain-field-clip-path)"
  >
    <path
      d="M 100 0 A426.66666666666674 426.66666666666674 0 0 1 0 222.22222 V266.66667 H100 Z"
      fill="#FF0000"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      stroke="#333"
      strokeWidth={1}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    />
    <path
      d="M 100 0 A426.66666666666674 426.66666666666674 0 0 0 200 222.22222 V266.66667 H100 Z"
      fill="#000000"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      stroke="#777"
      strokeWidth={1}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    />
  </g>
  <path
    d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
    fill="transparent"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    stroke="#333"
    strokeWidth={1}
    style={
      Object {
        "cursor": "pointer",
        "pointerEvents": "none",
      }
    }
  />
</svg>
`;

exports[`CoatsOfArms should render 'Barry azure and ermine' 1`] = `
<svg
  className="coats-of-arms-display"
  height={266.6666666666667}
  viewBox="0 0 200 266.6666666666667"
  width={200}
>
  <defs>
    <symbol
      id="ermine"
      viewBox="0 0 200 240"
    >
      <rect
        fill="#FFFFFF"
        height="100%"
        width="100%"
      />
      <g
        transform="translate(100, 240)"
      >
        <path
          d="M0,-174 C -15,-30 -29,-35 -55,0 L -15,-28 0,0 15,-28 55,0 C 28,-35 15,-30 0,-174 Z"
          fill="#000000"
        />
        <circle
          cy={-226}
          fill="#000000"
          r="14"
        />
        <circle
          cx={-32}
          cy={-194}
          fill="#000000"
          r={14}
        />
        <circle
          cx={32}
          cy={-194}
          fill="#000000"
          r={14}
        />
      </g>
    </symbol>
    <pattern
      height={64.93506493506494}
      id="field-pattern-3"
      patternUnits="userSpaceOnUse"
      viewBox="0 0 70 125"
      width={36.36363636363637}
    >
      <rect
        fill="#FFFFFF"
        height="125px"
        width="70px"
      />
      <use
        height={60}
        href="#ermine"
        width={35}
        x="0"
        y="0"
      />
      <use
        height={60}
        href="#ermine"
        width={35}
        x="35"
        y="65"
      />
    </pattern>
    
    
    
    
    
    
    
    
    
    
    
    <clipPath
      id="plain-field-clip-path"
    >
      <path
        d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
        fill="transparent"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        stroke="#333"
        strokeWidth={1}
        style={
          Object {
            "cursor": "pointer",
            "pointerEvents": "none",
          }
        }
      />
    </clipPath>
  </defs>
  <g
    className="blason-field"
    clipPath="url(#plain-field-clip-path)"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <pattern
      id="field-pattern-3-barry"
      patternTransform="scale(0.6833333333333332)"
      xlinkHref="#field-pattern-3"
    />
    <path
      d="M 0 -44.44444 H200 V44.44444 H0 Z"
      fill="#0000FF"
      stroke="#333"
    />
    <path
      d="M 0 44.44444 H200 V88.88889 H0 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 88.88889 H200 V133.33333 H0 Z"
      fill="#0000FF"
      stroke="#333"
    />
    <path
      d="M 0 133.33333 H200 V177.77778 H0 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 177.77778 H200 V222.22222 H0 Z"
      fill="#0000FF"
      stroke="#333"
    />
    <path
      d="M 0 222.22222 H200 V311.11111 H0 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
  </g>
  <path
    d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
    fill="transparent"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    stroke="#333"
    strokeWidth={1}
    style={
      Object {
        "cursor": "pointer",
        "pointerEvents": "none",
      }
    }
  />
</svg>
`;

exports[`CoatsOfArms should render 'Barry ermine and vert' 1`] = `
<svg
  className="coats-of-arms-display"
  height={266.6666666666667}
  viewBox="0 0 200 266.6666666666667"
  width={200}
>
  <defs>
    <symbol
      id="ermine"
      viewBox="0 0 200 240"
    >
      <rect
        fill="#FFFFFF"
        height="100%"
        width="100%"
      />
      <g
        transform="translate(100, 240)"
      >
        <path
          d="M0,-174 C -15,-30 -29,-35 -55,0 L -15,-28 0,0 15,-28 55,0 C 28,-35 15,-30 0,-174 Z"
          fill="#000000"
        />
        <circle
          cy={-226}
          fill="#000000"
          r="14"
        />
        <circle
          cx={-32}
          cy={-194}
          fill="#000000"
          r={14}
        />
        <circle
          cx={32}
          cy={-194}
          fill="#000000"
          r={14}
        />
      </g>
    </symbol>
    <pattern
      height={64.93506493506494}
      id="field-pattern-3"
      patternUnits="userSpaceOnUse"
      viewBox="0 0 70 125"
      width={36.36363636363637}
    >
      <rect
        fill="#FFFFFF"
        height="125px"
        width="70px"
      />
      <use
        height={60}
        href="#ermine"
        width={35}
        x="0"
        y="0"
      />
      <use
        height={60}
        href="#ermine"
        width={35}
        x="35"
        y="65"
      />
    </pattern>
    
    
    
    
    
    
    
    
    
    
    
    <clipPath
      id="plain-field-clip-path"
    >
      <path
        d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
        fill="transparent"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        stroke="#333"
        strokeWidth={1}
        style={
          Object {
            "cursor": "pointer",
            "pointerEvents": "none",
          }
        }
      />
    </clipPath>
  </defs>
  <g
    className="blason-field"
    clipPath="url(#plain-field-clip-path)"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <pattern
      id="field-pattern-3-barry"
      patternTransform="scale(0.6833333333333332)"
      xlinkHref="#field-pattern-3"
    />
    <path
      d="M 0 -44.44444 H200 V44.44444 H0 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 44.44444 H200 V88.88889 H0 Z"
      fill="#008000"
      stroke="#333"
    />
    <path
      d="M 0 88.88889 H200 V133.33333 H0 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 133.33333 H200 V177.77778 H0 Z"
      fill="#008000"
      stroke="#333"
    />
    <path
      d="M 0 177.77778 H200 V222.22222 H0 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 222.22222 H200 V311.11111 H0 Z"
      fill="#008000"
      stroke="#333"
    />
  </g>
  <path
    d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
    fill="transparent"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    stroke="#333"
    strokeWidth={1}
    style={
      Object {
        "cursor": "pointer",
        "pointerEvents": "none",
      }
    }
  />
</svg>
`;

exports[`CoatsOfArms should render 'Barry of six or and sable' 1`] = `
<svg
  className="coats-of-arms-display"
  height={266.6666666666667}
  viewBox="0 0 200 266.6666666666667"
  width={200}
>
  <defs>
    
    
    
    
    
    
    
    
    
    
    
    
    <clipPath
      id="plain-field-clip-path"
    >
      <path
        d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
        fill="transparent"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        stroke="#333"
        strokeWidth={1}
        style={
          Object {
            "cursor": "pointer",
            "pointerEvents": "none",
          }
        }
      />
    </clipPath>
  </defs>
  <g
    className="blason-field"
    clipPath="url(#plain-field-clip-path)"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <path
      d="M 0 -44.44444 H200 V44.44444 H0 Z"
      fill="#FFD700"
      stroke="#333"
    />
    <path
      d="M 0 44.44444 H200 V88.88889 H0 Z"
      fill="#000000"
      stroke="#333"
    />
    <path
      d="M 0 88.88889 H200 V133.33333 H0 Z"
      fill="#FFD700"
      stroke="#333"
    />
    <path
      d="M 0 133.33333 H200 V177.77778 H0 Z"
      fill="#000000"
      stroke="#333"
    />
    <path
      d="M 0 177.77778 H200 V222.22222 H0 Z"
      fill="#FFD700"
      stroke="#333"
    />
    <path
      d="M 0 222.22222 H200 V311.11111 H0 Z"
      fill="#000000"
      stroke="#333"
    />
  </g>
  <path
    d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
    fill="transparent"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    stroke="#333"
    strokeWidth={1}
    style={
      Object {
        "cursor": "pointer",
        "pointerEvents": "none",
      }
    }
  />
</svg>
`;

exports[`CoatsOfArms should render 'Barry wavy ermine and sable' 1`] = `
<svg
  className="coats-of-arms-display"
  height={266.6666666666667}
  viewBox="0 0 200 266.6666666666667"
  width={200}
>
  <defs>
    <symbol
      id="ermine"
      viewBox="0 0 200 240"
    >
      <rect
        fill="#FFFFFF"
        height="100%"
        width="100%"
      />
      <g
        transform="translate(100, 240)"
      >
        <path
          d="M0,-174 C -15,-30 -29,-35 -55,0 L -15,-28 0,0 15,-28 55,0 C 28,-35 15,-30 0,-174 Z"
          fill="#000000"
        />
        <circle
          cy={-226}
          fill="#000000"
          r="14"
        />
        <circle
          cx={-32}
          cy={-194}
          fill="#000000"
          r={14}
        />
        <circle
          cx={32}
          cy={-194}
          fill="#000000"
          r={14}
        />
      </g>
    </symbol>
    <pattern
      height={64.93506493506494}
      id="field-pattern-3"
      patternUnits="userSpaceOnUse"
      viewBox="0 0 70 125"
      width={36.36363636363637}
    >
      <rect
        fill="#FFFFFF"
        height="125px"
        width="70px"
      />
      <use
        height={60}
        href="#ermine"
        width={35}
        x="0"
        y="0"
      />
      <use
        height={60}
        href="#ermine"
        width={35}
        x="35"
        y="65"
      />
    </pattern>
    
    
    
    
    
    
    
    
    
    
    
    <clipPath
      id="plain-field-clip-path"
    >
      <path
        d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
        fill="transparent"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        stroke="#333"
        strokeWidth={1}
        style={
          Object {
            "cursor": "pointer",
            "pointerEvents": "none",
          }
        }
      />
    </clipPath>
  </defs>
  <g
    className="blason-field"
    clipPath="url(#plain-field-clip-path)"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <pattern
      id="field-pattern-3-barry"
      patternTransform="scale(0.6833333333333332)"
      xlinkHref="#field-pattern-3"
    />
    <path
      d="M 0 -44.44444 Q 5.55556 -39.11111 11.11111 -44.44444 Q 16.66667 -49.77778 22.22222 -44.44444 Q 27.77778 -39.11111 33.33333 -44.44444 Q 38.88889 -49.77778 44.44444 -44.44444 Q 50 -39.11111 55.55556 -44.44444 Q 61.11111 -49.77778 66.66667 -44.44444 Q 72.22222 -39.11111 77.77778 -44.44444 Q 83.33333 -49.77778 88.88889 -44.44444 Q 94.44444 -39.11111 100 -44.44444 Q 105.55556 -49.77778 111.11111 -44.44444 Q 116.66667 -39.11111 122.22222 -44.44444 Q 127.77778 -49.77778 133.33333 -44.44444 Q 138.88889 -39.11111 144.44444 -44.44444 Q 150 -49.77778 155.55556 -44.44444 Q 161.11111 -39.11111 166.66667 -44.44444 Q 172.22222 -49.77778 177.77778 -44.44444 Q 183.33333 -39.11111 188.88889 -44.44444 Q 194.44444 -49.77778 200 -44.44444 V44.44444 Q 194.44444 39.11111 188.88889 44.44444 Q 183.33333 49.77778 177.77778 44.44444 Q 172.22222 39.11111 166.66667 44.44444 Q 161.11111 49.77778 155.55556 44.44444 Q 150 39.11111 144.44444 44.44444 Q 138.88889 49.77778 133.33333 44.44444 Q 127.77778 39.11111 122.22222 44.44444 Q 116.66667 49.77778 111.11111 44.44444 Q 105.55556 39.11111 100 44.44444 Q 94.44444 49.77778 88.88889 44.44444 Q 83.33333 39.11111 77.77778 44.44444 Q 72.22222 49.77778 66.66667 44.44444 Q 61.11111 39.11111 55.55556 44.44444 Q 50 49.77778 44.44444 44.44444 Q 38.88889 39.11111 33.33333 44.44444 Q 27.77778 49.77778 22.22222 44.44444 Q 16.66667 39.11111 11.11111 44.44444 Q 5.55556 49.77778 0 44.44444 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 44.44444 Q 5.55556 49.77778 11.11111 44.44444 Q 16.66667 39.11111 22.22222 44.44444 Q 27.77778 49.77778 33.33333 44.44444 Q 38.88889 39.11111 44.44444 44.44444 Q 50 49.77778 55.55556 44.44444 Q 61.11111 39.11111 66.66667 44.44444 Q 72.22222 49.77778 77.77778 44.44444 Q 83.33333 39.11111 88.88889 44.44444 Q 94.44444 49.77778 100 44.44444 Q 105.55556 39.11111 111.11111 44.44444 Q 116.66667 49.77778 122.22222 44.44444 Q 127.77778 39.11111 133.33333 44.44444 Q 138.88889 49.77778 144.44444 44.44444 Q 150 39.11111 155.55556 44.44444 Q 161.11111 49.77778 166.66667 44.44444 Q 172.22222 39.11111 177.77778 44.44444 Q 183.33333 49.77778 188.88889 44.44444 Q 194.44444 39.11111 200 44.44444 V88.88889 Q 194.44444 83.55556 188.88889 88.88889 Q 183.33333 94.22222 177.77778 88.88889 Q 172.22222 83.55556 166.66667 88.88889 Q 161.11111 94.22222 155.55556 88.88889 Q 150 83.55556 144.44444 88.88889 Q 138.88889 94.22222 133.33333 88.88889 Q 127.77778 83.55556 122.22222 88.88889 Q 116.66667 94.22222 111.11111 88.88889 Q 105.55556 83.55556 100 88.88889 Q 94.44444 94.22222 88.88889 88.88889 Q 83.33333 83.55556 77.77778 88.88889 Q 72.22222 94.22222 66.66667 88.88889 Q 61.11111 83.55556 55.55556 88.88889 Q 50 94.22222 44.44444 88.88889 Q 38.88889 83.55556 33.33333 88.88889 Q 27.77778 94.22222 22.22222 88.88889 Q 16.66667 83.55556 11.11111 88.88889 Q 5.55556 94.22222 0 88.88889 Z"
      fill="#000000"
      stroke="#333"
    />
    <path
      d="M 0 88.88889 Q 5.55556 94.22222 11.11111 88.88889 Q 16.66667 83.55556 22.22222 88.88889 Q 27.77778 94.22222 33.33333 88.88889 Q 38.88889 83.55556 44.44444 88.88889 Q 50 94.22222 55.55556 88.88889 Q 61.11111 83.55556 66.66667 88.88889 Q 72.22222 94.22222 77.77778 88.88889 Q 83.33333 83.55556 88.88889 88.88889 Q 94.44444 94.22222 100 88.88889 Q 105.55556 83.55556 111.11111 88.88889 Q 116.66667 94.22222 122.22222 88.88889 Q 127.77778 83.55556 133.33333 88.88889 Q 138.88889 94.22222 144.44444 88.88889 Q 150 83.55556 155.55556 88.88889 Q 161.11111 94.22222 166.66667 88.88889 Q 172.22222 83.55556 177.77778 88.88889 Q 183.33333 94.22222 188.88889 88.88889 Q 194.44444 83.55556 200 88.88889 V133.33333 Q 194.44444 128 188.88889 133.33333 Q 183.33333 138.66667 177.77778 133.33333 Q 172.22222 128 166.66667 133.33333 Q 161.11111 138.66667 155.55556 133.33333 Q 150 128 144.44444 133.33333 Q 138.88889 138.66667 133.33333 133.33333 Q 127.77778 128 122.22222 133.33333 Q 116.66667 138.66667 111.11111 133.33333 Q 105.55556 128 100 133.33333 Q 94.44444 138.66667 88.88889 133.33333 Q 83.33333 128 77.77778 133.33333 Q 72.22222 138.66667 66.66667 133.33333 Q 61.11111 128 55.55556 133.33333 Q 50 138.66667 44.44444 133.33333 Q 38.88889 128 33.33333 133.33333 Q 27.77778 138.66667 22.22222 133.33333 Q 16.66667 128 11.11111 133.33333 Q 5.55556 138.66667 0 133.33333 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 133.33333 Q 5.55556 138.66667 11.11111 133.33333 Q 16.66667 128 22.22222 133.33333 Q 27.77778 138.66667 33.33333 133.33333 Q 38.88889 128 44.44444 133.33333 Q 50 138.66667 55.55556 133.33333 Q 61.11111 128 66.66667 133.33333 Q 72.22222 138.66667 77.77778 133.33333 Q 83.33333 128 88.88889 133.33333 Q 94.44444 138.66667 100 133.33333 Q 105.55556 128 111.11111 133.33333 Q 116.66667 138.66667 122.22222 133.33333 Q 127.77778 128 133.33333 133.33333 Q 138.88889 138.66667 144.44444 133.33333 Q 150 128 155.55556 133.33333 Q 161.11111 138.66667 166.66667 133.33333 Q 172.22222 128 177.77778 133.33333 Q 183.33333 138.66667 188.88889 133.33333 Q 194.44444 128 200 133.33333 V177.77778 Q 194.44444 172.44444 188.88889 177.77778 Q 183.33333 183.11111 177.77778 177.77778 Q 172.22222 172.44444 166.66667 177.77778 Q 161.11111 183.11111 155.55556 177.77778 Q 150 172.44444 144.44444 177.77778 Q 138.88889 183.11111 133.33333 177.77778 Q 127.77778 172.44444 122.22222 177.77778 Q 116.66667 183.11111 111.11111 177.77778 Q 105.55556 172.44444 100 177.77778 Q 94.44444 183.11111 88.88889 177.77778 Q 83.33333 172.44444 77.77778 177.77778 Q 72.22222 183.11111 66.66667 177.77778 Q 61.11111 172.44444 55.55556 177.77778 Q 50 183.11111 44.44444 177.77778 Q 38.88889 172.44444 33.33333 177.77778 Q 27.77778 183.11111 22.22222 177.77778 Q 16.66667 172.44444 11.11111 177.77778 Q 5.55556 183.11111 0 177.77778 Z"
      fill="#000000"
      stroke="#333"
    />
    <path
      d="M 0 177.77778 Q 5.55556 183.11111 11.11111 177.77778 Q 16.66667 172.44444 22.22222 177.77778 Q 27.77778 183.11111 33.33333 177.77778 Q 38.88889 172.44444 44.44444 177.77778 Q 50 183.11111 55.55556 177.77778 Q 61.11111 172.44444 66.66667 177.77778 Q 72.22222 183.11111 77.77778 177.77778 Q 83.33333 172.44444 88.88889 177.77778 Q 94.44444 183.11111 100 177.77778 Q 105.55556 172.44444 111.11111 177.77778 Q 116.66667 183.11111 122.22222 177.77778 Q 127.77778 172.44444 133.33333 177.77778 Q 138.88889 183.11111 144.44444 177.77778 Q 150 172.44444 155.55556 177.77778 Q 161.11111 183.11111 166.66667 177.77778 Q 172.22222 172.44444 177.77778 177.77778 Q 183.33333 183.11111 188.88889 177.77778 Q 194.44444 172.44444 200 177.77778 V222.22222 Q 194.44444 216.88889 188.88889 222.22222 Q 183.33333 227.55556 177.77778 222.22222 Q 172.22222 216.88889 166.66667 222.22222 Q 161.11111 227.55556 155.55556 222.22222 Q 150 216.88889 144.44444 222.22222 Q 138.88889 227.55556 133.33333 222.22222 Q 127.77778 216.88889 122.22222 222.22222 Q 116.66667 227.55556 111.11111 222.22222 Q 105.55556 216.88889 100 222.22222 Q 94.44444 227.55556 88.88889 222.22222 Q 83.33333 216.88889 77.77778 222.22222 Q 72.22222 227.55556 66.66667 222.22222 Q 61.11111 216.88889 55.55556 222.22222 Q 50 227.55556 44.44444 222.22222 Q 38.88889 216.88889 33.33333 222.22222 Q 27.77778 227.55556 22.22222 222.22222 Q 16.66667 216.88889 11.11111 222.22222 Q 5.55556 227.55556 0 222.22222 Z"
      fill="url(#field-pattern-3-barry)"
      stroke="#333"
    />
    <path
      d="M 0 222.22222 Q 5.55556 227.55556 11.11111 222.22222 Q 16.66667 216.88889 22.22222 222.22222 Q 27.77778 227.55556 33.33333 222.22222 Q 38.88889 216.88889 44.44444 222.22222 Q 50 227.55556 55.55556 222.22222 Q 61.11111 216.88889 66.66667 222.22222 Q 72.22222 227.55556 77.77778 222.22222 Q 83.33333 216.88889 88.88889 222.22222 Q 94.44444 227.55556 100 222.22222 Q 105.55556 216.88889 111.11111 222.22222 Q 116.66667 227.55556 122.22222 222.22222 Q 127.77778 216.88889 133.33333 222.22222 Q 138.88889 227.55556 144.44444 222.22222 Q 150 216.88889 155.55556 222.22222 Q 161.11111 227.55556 166.66667 222.22222 Q 172.22222 216.88889 177.77778 222.22222 Q 183.33333 227.55556 188.88889 222.22222 Q 194.44444 216.88889 200 222.22222 V311.11111 Q 194.44444 305.77778 188.88889 311.11111 Q 183.33333 316.44444 177.77778 311.11111 Q 172.22222 305.77778 166.66667 311.11111 Q 161.11111 316.44444 155.55556 311.11111 Q 150 305.77778 144.44444 311.11111 Q 138.88889 316.44444 133.33333 311.11111 Q 127.77778 305.77778 122.22222 311.11111 Q 116.66667 316.44444 111.11111 311.11111 Q 105.55556 305.77778 100 311.11111 Q 94.44444 316.44444 88.88889 311.11111 Q 83.33333 305.77778 77.77778 311.11111 Q 72.22222 316.44444 66.66667 311.11111 Q 61.11111 305.77778 55.55556 311.11111 Q 50 316.44444 44.44444 311.11111 Q 38.88889 305.77778 33.33333 311.11111 Q 27.77778 316.44444 22.22222 311.11111 Q 16.66667 305.77778 11.11111 311.11111 Q 5.55556 316.44444 0 311.11111 Z"
      fill="#000000"
      stroke="#333"
    />
  </g>
  <path
    d="M 0 0 H10 H190 H200 V160 Q 200 213.33333 100 266.66667 Q 0 213.33333 0 160 Z"
    fill="transparent"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    stroke="#333"
    strokeWidth={1}
    style={
      Object {
        "cursor": "pointer",
        "pointerEvents": "none",
      }
    }
  />
</svg>
`;
